channels/channel.streams/slam/chn_slam.py

```diff
@@ -0,0 +1,211 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import datetime
+
+from resources.lib import chn_class, mediatype
+from resources.lib.helpers.htmlentityhelper import HtmlEntityHelper
+
+from resources.lib.mediaitem import MediaItem
+from resources.lib.logger import Logger
+from resources.lib.helpers.jsonhelper import JsonHelper
+from resources.lib.urihandler import UriHandler
+from resources.lib.parserdata import ParserData
+from resources.lib.streams.m3u8 import M3u8
+
+
+class Channel(chn_class.Channel):
+    """
+    main class from which all channels inherit
+    """
+
+    def __init__(self, channel_info):
+        """Initialisation of the class.
+
+        Arguments:
+        channel_info: ChannelInfo - The channel info object to base this channel on.
+
+        All class variables should be instantiated here and this method should not
+        be overridden by any derived classes.
+
+        """
+
+        chn_class.Channel.__init__(self, channel_info)
+
+        # ==== Actual channel setup STARTS here and should be overwritten from derived classes =====
+        self.noImage = ""
+
+          # setup the urls
+        if self.channelCode == 'slam':
+            self.noImage = "slam.png"
+            self.mainListUri = "https://slam.nl"
+            self.baseUrl = "http://www.slam.nl"
+
+
+        # setup the main parsing data
+        self._add_data_parser(self.mainListUri, match_type=ParserData.MatchExact, json=True,
+                              preprocessor=self.add_live_streams)
+
+        # updater for live streams
+        self._add_data_parsers(["https://hls.slam.nl/streaming/hls/"],
+                               updater=self.update_live_stream_m3u8)
+
+        self._add_data_parser("https://playerservices.streamtheworld.com/api/livestream-redirect",
+                              updater=self.update_live_stream_redirect)
+
+        #===========================================================================================
+        # non standard items
+
+        #===========================================================================================
+        # Test cases:
+
+        #============================= Actual channel setup STOPS here =============================
+
+    def add_live_streams(self, data):
```

Comment:
This will break as soon as they change server. Better to use https://api.slam.nl/api/getStreams?brand=slam and use a JSON dataparser in Retrospect to fetch the actual live streams. 

The URL's they should then get look like: 

`https://playerservices.streamtheworld.com/api/livestream?station=web14&transports=http%2Chls%2Chlsts&version=1.9`

Reply:
You still don't use the API call to dynamically list the live streams. They are all hard coded now. Just make 'https://api.slam.nl/api/getStreams?brand=slam' the `self.mainListUri` and then make a parser that parses the URL in the correct live streams MediaItems.